packages/by-name/kata/kata-kernel-uvm/package.nix

```diff
@@ -2,11 +2,13 @@
 # SPDX-License-Identifier: AGPL-3.0-only
 
 {
+  lib,
   fetchurl,
   linuxManualConfig,
   stdenvNoCC,
   fetchzip,
   kata,
+  withGPU ? false,
```

Comment:
Why do we need a separate kernel for GPUs?

Reply:
This is aligned with how Kata handles this upstream. They build minimal kernels for each use-case, which is desirable from both a hardening perspective as well as the storage footprint.

Reply:
Is the diff actually that big?

Reply:
I wouldn't know why we should diverge from it, at least. 

And for the size of the diff: I can get you exact numbers, but I wouldn't expect it to be small. Consider driver-disablement (e.g. nouveau), PCI support, etc.

Reply:
Looking at the diff, we _could_ use the GPU config for both. The real diff is about 20 lines. (not counting moved options, etc.)

However, I'd still prefer to keep close to upstream, as this means that we don't have to keep up with the changes to both files on each Kata release. Do you have a strong opinion @burgerdev @katexochen ?

Reply:
I think what I missed is that the base config files are different for the two kernels - sorry.